### PR TITLE
Support for slots to handles crypto API changes under SPM

### PR DESCRIPTION
**Pull request by itayzafrir**
Support for the API changes from slots to handles under SPM (code separation between SPE and NSPE). 
Same idea as in https://github.com/ARMmbed/mbed-crypto/blob/development/library/psa_crypto.c#L29-L42

### CODE REVIEW

File: library/psa_crypto_slot_management.c [hunk 1]
```diff
@@ -26,6 +26,20 @@
 #endif
 
 #if defined(MBEDTLS_PSA_CRYPTO_C)
+/*
+ * In case MBEDTLS_PSA_CRYPTO_SPM is defined the code is built for SPM (Secure
+ * Partition Manager) integration which separate the code into two parts
+ * NSPE (Non-Secure Process Environment) and SPE (Secure Process Environment).
+ * In this mode an additional header file should be included.
```
Comment (Patater) [inline library/psa_crypto_slot_management.c]: Could we change this to the following?

```
When MBEDTLS_PSA_CRYPTO_SPM is defined, the code is being build for SPM (Secure Partition Manager) integration which separates the code into two parts: NSPE (Non-Secure Processing Element) and SPE (Secure Processing Element). When building for the SPE, an additional header file should be included.
```

File: library/psa_crypto_slot_management.c [hunk 2]
```diff
@@ -26,6 +26,20 @@
...
+ */
+#if defined(MBEDTLS_PSA_CRYPTO_SPM)
+/*
+ * PSA_CRYPTO_SECURE means that this file is compiled to the SPE side.
```
Comment (Patater) [inline library/psa_crypto_slot_management.c]:
```diff
- compiled to the SPE side.
- compiled for the SPE.
```

File: library/psa_crypto_slot_management.c [hunk 3]
```diff
@@ -26,6 +26,20 @@
...
+ * some headers will be affected by this flag.
```
Comment (Patater) [inline library/psa_crypto_slot_management.c]:
```diff
- some headers will be affected by this flag.
+ Some headers will be affected by this flag.
```

**Comment by Patater** [comment]
If you wouldn't mind, it'd help to have another commit to update the other comment block. Thanks.

File: library/psa_crypto_slot_management.c (hunk 1 above)
Comment (itayzafrir) [reply library/psa_crypto_slot_management.c]: Changed

File: library/psa_crypto_slot_management.c (hunk 2 above)
Comment (itayzafrir) [reply library/psa_crypto_slot_management.c]: Changed

File: library/psa_crypto_slot_management.c (hunk 3 above)
Comment (itayzafrir) [reply library/psa_crypto_slot_management.c]: Changed

**Comment by itayzafrir** [comment]
@Patater  I've updated the the code comments, commit message title and description, I also added a new commit to update existing SPM documentation in the code. 
Please take another look.

File: library/psa_crypto_slot_management.c (hunk 1 above)
Comment (Patater) [reply library/psa_crypto_slot_management.c]: I see this still says "Environment" instead of "Element".

Comment (gilles-peskine-arm) [reply library/psa_crypto_slot_management.c]: I don't see where “environment” should be changed to “element”. (N)SPE is (Non-)Secure Processing Environment.

**Pull request merged**